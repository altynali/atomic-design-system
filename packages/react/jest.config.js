/** @type {import('jest').Config} */

const config = {
  verbose: true,
  testEnvironment: "jsdom",
  collectCoverageFrom: ["src/**/*.{js,ts, tsx}"],
  roots: ["<rootDir>/src"],
}

module.exports = config
