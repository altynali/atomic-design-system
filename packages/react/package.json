{
  "name": "atomic-design-system-react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/node": "^20.10.0",
    "@types/react": "^18.2.23",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rollup": "^4.6.0",
    "rollup-plugin-typescript2": "^0.36.0",
    "typescript": "^5.3.2"
  },
  "dependencies": {},
  "scripts": {
    "build": "rollup -c --bundleConfigAsCjs",
    "dev": "yarn build --watch"
  }
}
