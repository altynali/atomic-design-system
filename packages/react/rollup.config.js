import Ts from "rollup-plugin-typescript2"

const config = {
  input: [
    "src/index.ts",
    "src/atoms/Color/index.ts",
    "src/atoms/Text/index.ts",
    "src/atoms/Margin/index.ts",
    "src/atoms/Button/index.ts",
    "src/atoms/Padding/index.ts",
    "src/atoms/Grid/index.ts",
    "src/atoms/Checkbox/index.ts",
    "src/molecules/Card/index.ts",
    "src/molecules/Select/index.ts",
    "src/molecules/Input/index.ts",
    "src/molecules/CardList/index.ts",
    "src/organisms/Form/index.ts",
    "src/templates/Layout/index.ts",
  ],
  output: [
    {
      dir: "lib",
      format: "esm",
      sourcemap: true,
      preserveModules: true,
    },
  ],
  plugins: [Ts()],
  external: ["react", "atomic-design-system-foundation"],
}

export default config
