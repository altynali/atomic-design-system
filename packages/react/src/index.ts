import { Color } from "./atoms/Color"
import { Text } from "./atoms/Text"
import { Margin } from "./atoms/Margin"
import { Button } from "./atoms/Button"
import { Select } from "./molecules/Select"

export { Color, Text, Margin, Button, Select }

// TODO: import * from './atoms';
