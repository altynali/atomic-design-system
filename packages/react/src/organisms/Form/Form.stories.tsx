import React from "react"
import Form from "./Form"
// import "atomic-design-system-scss/lib/Form.css"
import "atomic-design-system-scss/lib/Button.css"
import "atomic-design-system-scss/lib/Input.css"
import "atomic-design-system-scss/lib/Select.css"

export default {
  title: "Organisms|Form",
}

export const Common = () => <Form />
