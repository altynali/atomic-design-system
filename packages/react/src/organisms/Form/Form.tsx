import React, { FC, PropsWithChildren, FormEvent } from "react"
import { Button } from "../../atoms/Button"
import { Text } from "../../atoms/Text"
import { Padding } from "../../atoms/Padding"
import { Spacing } from "atomic-design-system-foundation"
import { classNames } from "../../utils"

export type FormProps = {
  className?: string
  label?: string
  onSubmit?: () => void
} & PropsWithChildren

const Form: FC<FormProps> = (props) => {
  const { children, label, onSubmit, className = "" } = props
  const idForm = `form-${label}`

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault()
    onSubmit?.()
  }

  return (
    <form
      className={classNames(className, ["atds-form"])}
      onSubmit={(event: FormEvent) => handleSubmit(event)}
      aria-label={idForm}
      data-testid="AtdsForm"
    >
      {label && (
        <Padding bottom space={Spacing.sm}>
          <label htmlFor={idForm} id={idForm}>
            <Text>{label}</Text>
          </label>
        </Padding>
      )}
      {children}
      <Button type="submit">Submit</Button>
    </form>
  )
}

export default Form
