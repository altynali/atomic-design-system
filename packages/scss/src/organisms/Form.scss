@import "foundation/all";

.atds-form {
  width: 100%;
}
