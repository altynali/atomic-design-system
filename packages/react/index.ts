import { Color, Margin, Select, Text } from "./src/index"

export { Color, Text, Margin, Select }

// TODO: import * from './atoms';
