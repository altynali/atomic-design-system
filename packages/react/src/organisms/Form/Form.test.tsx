import React from "react"
import { render, fireEvent } from "@testing-library/react"
import Form from "./Form"
import "@testing-library/jest-dom"

describe("Form component", () => {
  it("renders form with label and children", () => {
    const { getByTestId } = render(
      <Form label="Test Form" onSubmit={() => {}}>
        <input type="text" />
      </Form>
    )
    const form = getByTestId("AtdsForm")
    expect(form).toBeInTheDocument()
  })

  it("applies custom className", () => {
    const { container } = render(<Form className="custom" />)
    const form = container.querySelector("form")
    expect(form).toHaveClass("custom")
  })

  it("calls onSubmit event handler when submitted", () => {
    const onSubmitMock = jest.fn()
    const { getByText } = render(<Form onSubmit={onSubmitMock} />)
    const submitButton = getByText("Submit")
    fireEvent.click(submitButton)
    expect(onSubmitMock).toHaveBeenCalledTimes(1)
  })

  test("snapshot of the base state", () => {
    const { asFragment } = render(<Form />)

    expect(asFragment()).toMatchSnapshot()
  })
})
