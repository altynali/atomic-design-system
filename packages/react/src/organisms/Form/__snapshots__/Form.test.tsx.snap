// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot of the base state 1`] = `
<DocumentFragment>
  <form
    aria-label="form-undefined"
    class=" atds-form"
    data-testid="AtdsForm"
  >
    <button
      class=" atds-button-container atds-button-base"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
